packages/atlas/src/components/_comments/CommentInput/CommentInput.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useEffect, useRef, useState } from 'react'
+import mergeRefs from 'react-merge-refs'
+import useResizeObserver from 'use-resize-observer'
+
+import { Information } from '@/components/Information'
+import { Text } from '@/components/Text'
+import { Button } from '@/components/_buttons/Button'
+import { useMediaMatch } from '@/hooks/useMediaMatch'
+
+import { Border, ButtonsContainer, Container, Flex, StyledCommentRow, StyledTextArea } from './CommentInput.styles'
+
+import { CommentRowProps } from '../CommentRow'
+
+export type CommentInputProps = {
+  processing: boolean
+  onComment?: () => void
+  onCancel?: () => void
+} & CommentRowProps
+
+export const CommentInput: React.FC<CommentInputProps> = ({ processing, onCancel, onComment, ...rest }) => {
+  const smMatch = useMediaMatch('sm')
+  const containerRef = useRef<HTMLDivElement>(null)
+  const textAreaRef = useRef<HTMLTextAreaElement>(null)
+  const [active, setActive] = useState(false)
+  const [text, setText] = useState('')
+
+  const { ref: measureRef, height: textAreaHeight = 40 } = useResizeObserver({ box: 'border-box' })
+
+  useEffect(() => {
+    if (active) textAreaRef.current?.focus()
+  }, [active])
+
+  useEffect(() => {
+    if (!active) {
+      return
+    }
+    const handleClickOutside = (event: Event) => {
+      if (containerRef.current && !containerRef.current.contains(event.target as Node)) {
+        // stop propagation so it doesn't get triggered again on button click
+        // prevent default so it doesn't trigger unwanted submits
+        event.preventDefault()
+        event.stopPropagation()
+        setActive(false)
+      }
+    }
+    document.addEventListener('click', handleClickOutside, true)
+    return () => {
+      document.removeEventListener('click', handleClickOutside, true)
+    }
+  }, [active])
+
+  const onChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => setText(e.currentTarget.value)
+
+  const show = !!text || active
+
+  return (
+    <StyledCommentRow {...rest}>
+      <Container
+        ref={containerRef}
+        data-show={show}
+        height={textAreaHeight}
+        onKeyDown={(e) => {
+          if ((e.nativeEvent.ctrlKey || e.nativeEvent.metaKey) && e.nativeEvent.code === 'Enter') {
+            onComment?.()
+          }
+        }}
+        onClick={() => {
+          setActive(true)
+          textAreaRef.current?.focus()
```

Comment:
Correct me if I'm wrong but I'm pretty sure that you don't need this. If Container had been a label, the text area would just focus automatically.

Reply:
yeah, we don't need focus here but we do need to setActive sadly